### PR TITLE
SSCS-10223 gaps override changes

### PR DESCRIPTION
**Pull request by wmarshid**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/SSCS-10223

### Change description ###

Adds a message handler to the "ResendToGaps" event listener to add a "cancel hearing" message onto the Hearings Event Queue.

Re-created from #2638 

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/resendtogaps/ResendToGapsAboutToSubmitHandler.java [hunk 1]
```diff
@@ -21,26 +23,22 @@
...
 import uk.gov.hmcts.reform.sscs.robotics.RoboticsWrapper;
 
-@Component
 @Slf4j
+@Component
+@RequiredArgsConstructor
 public class ResendToGapsAboutToSubmitHandler implements PreSubmitCallbackHandler<SscsCaseData> {
 
-    private PreSubmitCallbackResponse<SscsCaseData> preSubmitCallbackResponse;
+    @Value("${feature.gaps-switchover.enabled:false}")
```
Comment (dattipoeHMCTS) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/resendtogaps/ResendToGapsAboutToSubmitHandler.java:31]: Does the : notation work here?

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/resendtogaps/ListAssistHearingMessageHelper.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+import uk.gov.hmcts.reform.sscs.model.hearings.HearingRequest;
+import uk.gov.hmcts.reform.sscs.service.servicebus.HearingMessagingServiceFactory;
+
+@Service
+@RequiredArgsConstructor
+public class ListAssistHearingMessageHelper {
+
+    private final HearingMessagingServiceFactory hearingMessagingServiceFactory;
+
+    public void sendListAssistCancelHearingMessage(final String ccdCaseId) {
```
Comment (dattipoeHMCTS) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/resendtogaps/ListAssistHearingMessageHelper.java:17]: I think it'd be better to pass in the required arguments so this method can be something like "sendMessage" - making it more generic

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/resendtogaps/ResendToGapsAboutToSubmitHandler.java (hunk 1 above)
Comment (wmarshid) [reply src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/resendtogaps/ResendToGapsAboutToSubmitHandler.java:31]: should do.. how do you guys do it?

Comment (dattipoeHMCTS) [reply src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/resendtogaps/ResendToGapsAboutToSubmitHandler.java:31]: It's usually just in the application.yaml file

Comment (wmarshid) [reply src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/resendtogaps/ResendToGapsAboutToSubmitHandler.java:31]: okay thanks I'll change it

**Review by dattipoeHMCTS — APPROVED** [review]

**Review by dattipoeHMCTS — APPROVED** [review]

File: src/test/resources/json/sya_with_ccdId.json [hunk 3]
```diff
@@ -5,7 +5,7 @@
   },
   "mrn": {
     "dwpIssuingOffice": "DWP PIP (1)",
-    "date": "01-04-2021",
+    "date": "01-04-2022",
```
Comment (TeaWithLucas) [inline src/test/resources/json/sya_with_ccdId.json:8]: change this to a distant date, e.g. 2999 to avoid future problems

**Review by TeaWithLucas — APPROVED** [review]

**Pull request merged**